### PR TITLE
boards: add a dedicated Qemu variant based on MPS2_AN385

### PR DESCRIPTION
**Pull request by nashif**
This is the result of issues we had in #10556 related to running in Qemu.

Edit by @pfalcon: specific pointer: https://github.com/zephyrproject-rtos/zephyr/pull/10556#discussion_r232410695

### CODE REVIEW

**Review by andyross — APPROVED** [review]
This seems better.  FWIW: you could eliminate some of the pasted hardware stuff, as AFAIK there's no qemu emulation support for stuff like gpio and i2c (but... maybe I'm wrong, no idea).

**Review by galak — DISMISSED** [review]
Rather than duplicating this, why not just add a second defconfig, and Kconfig to the MPS2_AN385 for qemu.

**Comment by nashif** [comment]
> This seems better. FWIW: you could eliminate some of the pasted hardware stuff, as AFAIK there's no qemu emulation support for stuff like gpio and i2c (but... maybe I'm wrong, no idea).

fixed now

**Comment by nashif** [comment]
> Rather than duplicating this, why not just add a second defconfig, and Kconfig to the MPS2_AN385 for qemu.

It is not going to be a duplication, all files have been changed for the qemu target now, this is going to be similar to qemu_riscv32

**Comment by nashif** [comment]
@galak please revisit

**Comment by nashif** [comment]
@galak ping

File: boards/arm/qemu_mps2/qemu_mps2.dts [hunk 1]
```diff
@@ -0,0 +1,107 @@
+/dts-v1/;
+
+#include <arm/armv7-m.dtsi>
+
+/ {
```
Comment (pfalcon) [inline boards/arm/qemu_mps2/qemu_mps2.dts]: So, why we duplicate all this, instead of #inlcude'ing it?

Comment (nashif) [reply boards/arm/qemu_mps2/qemu_mps2.dts]: including it from where?

Comment (pfalcon) [reply boards/arm/qemu_mps2/qemu_mps2.dts]: From real mps_an385 DTS.

Comment (nashif) [reply boards/arm/qemu_mps2/qemu_mps2.dts]: it is not an exact copy, we have removed anything that does not apply to running in qemu.

Comment (pfalcon) [reply boards/arm/qemu_mps2/qemu_mps2.dts]: Well, seems like we're just eager to complicate our own life here. If the only (?) problem with mps2_an385 vs qemu variant of it is the need to disable tickless mode, why do more work than that?

**Review by tyrantlizard77 — APPROVED** [review]

**Pull request closed without merging**